bot.js

```diff
@@ -33,11 +33,21 @@ discordClient.on('messageReactionAdd', async (reaction) => {
 				}
 			});
 		} else if (reaction.emoji.name == 'deleteThis') {
-			console.log(`Attempting to delete the tweet referenced in "${reaction.message.content}".`)
-			const tweetId = lib.ExtractTweetId(reaction.message.content)
-			await lib.DeleteTweet(twitterClient, tweetId);
-			console.log(`Tweet ${tweetId} deleted.`);
-			reaction.message.lineReplyNoMention("I deleted that tweet; please don't get Jimmy cancelled.")
+			await handleOneReactionAtATime.runExclusive(async () => {
+				if (await lib.NoTweetFound(records, reaction)) {
+					console.log(`Attempting to delete the tweet referenced in "${reaction.message.content}".`)
+					const tweetId = lib.ExtractTweetId(reaction.message.content)
+					await lib.DeleteTweet(twitterClient, tweetId);
+					console.log(`Tweet ${tweetId} deleted.`);
+					reaction.message.lineReplyNoMention("I deleted that tweet; please don't get Jimmy cancelled.")
+				} else {
+					var tweetId = await lib.GetKnownTweetUrl(records, reaction);
+					console.log(`Attempting to delete the tweet referenced in "${reaction.message.content}".`)
```

Comment:
Lots of code duplicated, everything is the same after you fetch the `tweetId`.

```
if (await lib.NoTweetFound(records, reaction)) {
    const tweetId = lib.ExtractTweetId(reaction.message.content);
} else {
   const tweetId = await lib.GetKnownTweetUrl(records, reaction);
}

// everything else..
```

Reply:
Here's the new deleteThis

```suggestion
		} else if (reaction.emoji.name == 'deleteThis') {
			await handleOneReactionAtATime.runExclusive(async () => {
				if (await lib.NoTweetFound(records, reaction)) {
					var tweetId = lib.ExtractTweetId(reaction.message.content)
				} else {
					var tweetId = await lib.GetKnownTweetUrl(records, reaction);
				}
				await lib.DeleteTweet(twitterClient, tweetId);
				console.log(`Tweet ${tweetId} deleted.`);
				reaction.message.lineReplyNoMention("I deleted that tweet; please don't get Jimmy cancelled.")
			})
		}
	}
```